languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -35762,6 +35762,88 @@ though, and has since been slightly modified)-->
     </rule>
   </category>
   <category id="CAT_TOURS_CRITIQUES" name="Tours critiqués divers (barbarismes, impropriétés, solécismes, etc.)">
+      <rulegroup id="IL_Y_A" name="il y a" default="temp_off">
+          <rule>
+              <antipattern>
+                  <token>n'</token>
+                  <token>y</token>
+                  <token>a</token>
+              </antipattern>
+              <antipattern>
+                  <token>y</token>
+                  <token>a</token>
+                  <token>-t-il</token>
+              </antipattern>
+              <antipattern>
+                  <token postag="[RNZ].*" postag_regexp="yes"></token>
+                  <token>y</token>
+                  <token>a</token>
+              </antipattern>
+              <pattern>
+                  <marker>
+                      <token>y</token>
+                      <token>a<exception scope="next">t</exception></token>
+                  </marker>
+              </pattern>
+              <message>Cette expression est considérée comme familière.</message>
+              <suggestion>il \1 \2</suggestion>
+              <example correction="il y a">Selon lui, <marker>y a</marker> un problème.</example>
+          </rule>
+          <rule>
```

Comment:
Please make sure the first rule doesn't steal from the second rule. I.e., the first needs an AP for pas/plus etc.

Reply:
I added the AP required and improved the rule.